src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -114,7 +114,8 @@ static SingleValue GetTypeValueNodeFromGenericArgument (ITypeSymbol type)
 				// That said we only use it to perform the dynamically accessed members checks and for that purpose treating it as System.Type is perfectly valid.
 				SymbolKind.NamedType => new SystemTypeValue ((INamedTypeSymbol) type),
 				SymbolKind.ErrorType => UnknownValue.Instance,
-				// What about things like ArrayType or PointerType and so on. Linker treats these as "named types" since it can resolve them to concrete type
+				SymbolKind.ArrayType => new ArrayValue (GetTypeValueNodeFromGenericArgument (((IArrayTypeSymbol) type).ElementType)),
```

Comment:
I think it doesn't make sense to call `GetTypeValueNodeFromGenericArgument` on the element type - the element type isn't the generic argument.

Reply:
In this case it also doesn't make sense to use `ArrayValue`. The generic parameter is a type, not a value. So even the result of typeof<int[]> is not an array value, it's still a type (`SystemTypeValue`).

Reply:
In linker what this does (roughly) is to mark the item type of the array as a type (so nothing on it) - this operation is something I don't know yet if it needs to be handled by the analyzer (the only case where it might matter is type hierarchy marking... but I didn't look into the details on it yet). The resulting type which we then track through data flow is just the basic `System.Array`, since that's what one would actually reflect on if this were used in reflection.